Editor/Script/EmptyMethodAnalyzer.cs

```diff
@@ -0,0 +1,57 @@
+using System.Collections.Generic;
+using Mono.Cecil;
+using Mono.Cecil.Cil;
+using Unity.ProjectAuditor.Editor.Utils;
+
+namespace Unity.ProjectAuditor.Editor
+{
+    [ScriptAnalyzer]
+    public class EmptyMethodAnalyzer : IInstructionAnalyzer
+    {
+        private static readonly ProblemDescriptor descriptor = new ProblemDescriptor
+        {
+            id = 10201,
+            description = "Empty Method",
+            type = string.Empty,
+            method = string.Empty,
+            area = "CPU",
+            problem = "Any empty Awake/Start/Update/LateUpdate/FixedUpdate will be included in the build and executed anyway.",
+            solution = "Remove any empty MonoBehaviour methods."
+        };
+
+        public static ProblemDescriptor GetDescriptor()
+        {
+            return descriptor;
+        }
+        
+        private OpCode[] m_OpCodes = new[] {OpCodes.Ret};
+        
+        public EmptyMethodAnalyzer(ScriptAuditor auditor)
+        {
+            auditor.RegisterDescriptor(descriptor);
+        }
+        
+        public ProjectIssue Analyze(MethodDefinition methodDefinition, Instruction inst)
+        {
+            if (inst.Previous != null)
```

Comment:
>this will miss method bodies with nops and a ret
out of curiosity, what would be an example of a method that results in nop+ret? Aside from something that you specifically generate with something like via Mono Cecil.

Would it be enough to skip all 'nops', if any, before checking whether 'ret' is the only instruction?

Reply:
> what about abstract methods?
abstract methods are skipped because they have no body.

Reply:
sure, just wonder if you are processing *Bar.Update()*

Reply:
> Would it be enough to skip all 'nops', if any, before checking whether 'ret' is the only instruction?
As long as there are no other instructions other than *nops* I guess so.